Fixup github action to install and run tests

lparsons opened the pull request:
Currently, `test_5` and `test_25` fail. They are the two tests with gzipped output, and I suspect there is a problem testing that output (or perhaps creating it as well).

lparsons committed:
Fixup github action to install and run tests

Currently, test_5 fails

hepcat72 commented on .github/workflows/setup_perl_env.yml:
```diff
@@ -4,26 +4,38 @@ jobs:
+    - name: build and install htseq2multx
+      run: |
+        cd htseq2multx
+        perl Makefile.PL
+        make
+        make install
```
I was wondering if I needed this or if cpanm took care of it, because the script was running...  I had assumed I would need these steps, but when I saw it spitting out the usage, I figured cpanm may have handled it.

hepcat72 commented on .github/workflows/setup_perl_env.yml:
```diff
@@ -4,26 +4,38 @@ jobs:
-    - uses: actions/checkout@v2
-    - name: Setup perl & install htseq2multx
+        chmod +x fastq-multx
+        echo "::add-path::${GITHUB_WORKSPACE}/fastq-multx"
```
Yeah, I wasn't sure I needed the mkdir and the cp, but I had tried adding the fastq-multx directory to the past using a relative path like in the docs, but it wasn't finding it.  When I found the `GITHUB_WORKSPACE` variable, I saw it in an example that used those mkdir/cp commands.  That example was using a container though.

So I'm wondering why this works... The significant differences of your changes are the htseq2multx build commands.  Do you think that's what did it?

hepcat72 commented on .github/workflows/setup_perl_env.yml:
```diff
@@ -4,26 +4,38 @@ jobs:
+    - name: install htseq2multx dependencies
+      run: |
+        cd htseq2multx
         cpanm --installdeps --notest .
```
So I'm curious about this.  I was seeing the tests run twice: once from the cpanm command (despite the --notests option, and once from my shell script call.  I couldn't understand why that was happening, but I was more focussed on the fastq-multx missing issue.

hepcat72 committed:
Tried removing clean

Clean may not be necessary.
I also edited step names.

hepcat72 committed:
Adding debug arg to tests

Going to see if the debug output can explain why tests 5 & 25 fail.  It looks like those are the only tests where line 42 (containing gzip in a process substitution call) executes, so I figure that must have something to do with it.

hepcat72 committed:
Debug level 3

This adds set -x.
Fixed a name typo.

hepcat72 committed:
Seeing if commenting this code fixes tests 5 & 25

These tests end up generating a broken stdout pipe error from gzip, which must be coming from fastq-multx.  The tests only fail in the github action, so I'm committing this commenting of those lines to test to see if it resolves the issue.

hepcat72 committed:
Removing debug arg

hepcat72 committed:
Test omission of gzip broken pipe error

The error doesn't seem to have a deleterious effect and only occurs on certain systems.

hepcat72 committed:
Work-around for gzip stdout broken pipe error on some systems
